crates/chain/src/indexer/keychain_txout.rs

```diff
@@ -147,7 +147,7 @@ impl<K: Clone + Ord + Debug> Indexer for KeychainTxOutIndex<K> {
             if self.last_revealed.get(did) < Some(&index) {
                 self.last_revealed.insert(*did, index);
                 changeset.last_revealed.insert(*did, index);
-                self.replenish_inner_index(*did, &keychain, self.lookahead);
+                self.replenish_inner_index(*did, keychain, self.lookahead);
```

Comment:
Is this just removing an additional `&` since the `.scan_txout(I) -> &I` (returns a reference) and `keychain` here is already a reference?

Reply:
You are right I will fix that

Reply:
A little higher up in the code (line 142) we've `cloned()` the result of `scan_txout(tx)` so by line 150 it's already an owned K. 
 
But bigger picture, since `replenish_inner_index()` is an internal API an owned K maybe isn't strictly needed here, but since we're making the assumption K is always small I think it's OK to do for consistency.

Reply:
Oops, I see @rustaceanrob took the other approach of changing `replenish_inner_index()` I'm fine with going that way too.

Reply:
Oh shoot I already changed the inner functions to use a `&K` if there was unneeded clones in my rebase. I agree that the consistency is a nice to have, but since the `replenish_inner_index` logic probably won't change I feel like we might as well take a reference?

Reply:
Yup it's fine that you changed `replenish_inner_index` to take a ref.

Reply:
I think it's fine to pass in an owned value of `K` since it's cloned in the function body anyway